homeassistant/components/automation/time.py

```diff
@@ -13,20 +13,29 @@
 _LOGGER = logging.getLogger(__name__)
 
 TRIGGER_SCHEMA = vol.Schema(
-    {vol.Required(CONF_PLATFORM): "time", vol.Required(CONF_AT): cv.time}
+    {
+        vol.Required(CONF_PLATFORM): "time",
+        vol.Required(CONF_AT): vol.All(cv.ensure_list, [cv.time]),
+    }
 )
 
 
 async def async_attach_trigger(hass, config, action, automation_info):
     """Listen for state changes based on configuration."""
-    at_time = config.get(CONF_AT)
-    hours, minutes, seconds = at_time.hour, at_time.minute, at_time.second
+    at_times = config[CONF_AT]
 
     @callback
     def time_automation_listener(now):
         """Listen for time changes and calls action."""
         hass.async_run_job(action, {"trigger": {"platform": "time", "now": now}})
 
-    return async_track_time_change(
-        hass, time_automation_listener, hour=hours, minute=minutes, second=seconds
-    )
+    return [
```

Comment:
note to self: Need double check this can handle an array of unsub/removes

Reply:
Looks like that is an issue

```
2020-07-21 17:21:31 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.140505127388240] 'list' object is not callable
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 130, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1250, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1285, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 518, in admin_handler
    await result
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 226, in reload_service_handler
    conf = await component.async_prepare_reload()
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 300, in async_prepare_reload
    await self._async_reset()
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 262, in _async_reset
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 469, in async_reset
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 485, in async_remove_entity
    await self.entities[entity_id].async_remove()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 494, in async_remove
    await self.async_will_remove_from_hass()
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 416, in async_will_remove_from_hass
    await self.async_disable()
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 455, in async_disable
    self._async_detach_triggers()
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 491, in remove_triggers
    remove()
TypeError: 'list' object is not callable
```

Reply:
Resolved it with the suggestion Paulus made below.